src/cancelq/posix/cancellable.c

```diff
@@ -0,0 +1,510 @@
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <stdio.h>
+#include <pthread.h>
+
+#include "cancellable.h"
+
+#define HEAP_CHECK true // Set to false to save oodles of execution time.
+
```

Comment:
Maybe wrap this in an `#ifdef DEBUG` stanza so it can be controlled by the build?

Reply:
Good idea